modules/compute/gke-node-pool/versions.tf

```diff
@@ -18,11 +18,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 5.0"
+      version = "> 5.0"
```

Comment:
nit: I think `>= 5` is a safer way of expressing this idea in general. I believe a version named `5.0` would not pass your test, but `5.0.0` would.